src/main/java/org/candlepin/subscriptions/tally/SnapshotSummaryProducer.java

```diff
@@ -61,19 +65,32 @@ protected SnapshotSummaryProducer(
   public void produceTallySummaryMessages(Map<String, List<TallySnapshot>> newAndUpdatedSnapshots) {
     AtomicInteger totalTallies = new AtomicInteger();
     newAndUpdatedSnapshots.forEach(
-        (orgId, snapshots) ->
-            snapshots.stream()
-                .sorted(Comparator.comparing(TallySnapshot::getSnapshotDate))
-                .map(snapshot -> summaryMapper.mapSnapshots(orgId, List.of(snapshot)))
-                .forEach(
-                    summary -> {
-                      if (validateTallySummary(summary)) {
-                        kafkaRetryTemplate.execute(
-                            ctx ->
-                                tallySummaryKafkaTemplate.send(tallySummaryTopic, orgId, summary));
-                        totalTallies.getAndIncrement();
-                      }
-                    }));
+        (orgId, snapshots) -> {
+          // Filter snapshots, as we only deal with hourly and non Any fields when we transmit the
+          // tally summary message to the BillableUsage component.
+          List<TallySnapshot> filteredSnapshots =
+              snapshots.stream()
+                  .filter(
+                      snapshot ->
+                          snapshot.getGranularity().equals(Granularity.HOURLY)
+                              && !snapshot.getServiceLevel().equals(ServiceLevel._ANY)
+                              && !snapshot.getUsage().equals(Usage._ANY)
+                              && !snapshot.getBillingProvider().equals(BillingProvider._ANY)
+                              && !snapshot.getBillingAccountId().equals("_ANY"))
+                  .sorted(Comparator.comparing(TallySnapshot::getSnapshotDate))
+                  .toList();
```

Comment:
Why do you need to build the list to loop for the items?
Another interesting idea would be to change the `validateTallySummary` method which as far as I see, it's only checking if the snapshot has measurements. So, renaming this method to `hasMeasurements` that accepts a tally snapshot instead, so we don't unnecessary map the pojo from snapshot to summary.

To sum up, the `produceTallySummaryMessages` method would look like as:
```
snapshots.stream()
                  .filter(byHourlyAndNotAnyFields())
                  .filter(hasMeasurements())
                  .sorted(Comparator.comparing(TallySnapshot::getSnapshotDate))
                  .map(toTallySummary())
                  .forEach(sendTallySummary());
```

Reply:
Sounds good suggestion. I will take a look on refactoring it.

Reply:
The ValidateTallysummary did one more step of  .findFirst() that removed either Total or Physical depending on what is first so we don't send both. My code does an additional step by peeking and removing any measurements with Total.